src/Semaphore.sol

```diff
@@ -139,8 +162,8 @@ contract Semaphore is IWorldID, SemaphoreCore, Verifier, SemaphoreGroups {
         RootHistory memory rootData = rootHistory[root];
 
         if (
-            rootData.groupId != groupId ||
-            block.timestamp - rootData.timestamp > ROOT_HISTORY_EXPIRY
+                rootData.groupId != groupId ||
+                (root != latestRoots[groupId] && block.timestamp - rootData.timestamp > ROOT_HISTORY_EXPIRY)
         ) revert InvalidRoot();
```

Comment:
This seems duplicate with `checkValidRoot` can we replace it with a call to that function?

Reply:
I would like to make this a modifier instead, just avoided doing to keep the pattern but should give a nominal gas saving.